packages/core/src/reducers/activities.js

```diff
@@ -25,9 +25,14 @@ function findByClientActivityID(clientActivityID) {
 function upsertActivityWithSort(activities, nextActivity) {
   const {
     channelData: { clientActivityID: nextClientActivityID } = {},
-    from: { id: nextFromID } = {}
+    from: { id: nextFromID, role: nextFromRole } = {},
+    type: nextType
   } = nextActivity;
 
+  if (nextType === 'typing' && nextFromRole === 'user') {
```

Comment:
When Direct Line platform echoed back the user typing activity, we should not add it because it was not intended to show up in the UI.